src/coreclr/vm/callhelpers.cpp

```diff
@@ -515,6 +515,9 @@ void MethodDescCallSite::CallTargetWorker(const ARG_SLOT *pArguments, ARG_SLOT *
     CallDescrData callDescrData;
 
     callDescrData.pSrc = pTransitionBlock + sizeof(TransitionBlock);
+#ifdef TARGET_WASM
+    callDescrData.pTransitionBlock = (TransitionBlock*)pTransitionBlock;
```

Comment:
What about `DispatchCallSimple` earlier in this file? It builds `CallDescrData`` as well, but none of these extra fields are initialized.

I agree with some of the earlier comments that the calling convention handling does not look very clean in this PR.

I am maybe better to leave it to for a separate PR (or wrapped with a bunch of TODOs to make it clear where we are doing temporary throw-away changes).

Reply:
I have put an assert in the DispatchCallSimple and updated other parts based on the feedback. Could it go in like this or do you prefer to extract parts (which?) of this PR to extra PR?

Reply:
I think I can live with what's in this PR, but we should avoid growing it organically. We want to have discussion about how the calling convention should actually work before doing more work on it here.

> this PR vs. extra PR

This PR has a bunch of good changes and a bunch of changes that need more work. If you would like to, I can go over it with you over it and separate the good non-controversial changes and get those in, and then work on the rest separately.